File: VegtablityApi/app/schemas/shift.py
from pydantic import BaseModel
from typing import Optional
from datetime import datetime

class ShiftOpenRequest(BaseModel):
    StartingCash: float

class ShiftResponse(BaseModel):
    ShiftID: int
    UserID: int
    StartTime: datetime
    StartingCash: float
    Status: str

class ShiftSummaryResponse(BaseModel):
    ShiftID: int
    UserID: int
    UserName: Optional[str] = None
    StartTime: datetime
    EndTime: Optional[datetime] = None
    StartingCash: float
    EndingCash: Optional[float] = None
    Status: str
    TotalSales: float
    TotalPurchases: float
    SalesCount: int
    PurchasesCount: int
    TotalPaidSales: float
    TotalCashSales: Optional[float] = 0.0
    CashSales: Optional[float] = 0.0
    TotalKnetSales: Optional[float] = 0.0
    KnetSales: Optional[float] = 0.0
    TotalNonCashSales: Optional[float] = 0.0
    CardSales: Optional[float] = 0.0
    TotalRemainder: float
    TotalPaidPurchases: float
    TotalCashPurchases: Optional[float] = 0.0
    TotalNonCashPurchases: Optional[float] = 0.0
    TotalPurchasesRemainder: float
    TotalReceiptVouchers: Optional[float] = 0.0
    TotalPaymentVouchers: Optional[float] = 0.0
    TotalExpenses: Optional[float] = 0.0
    ExpectedCash: Optional[float] = 0.0
    Difference: Optional[float] = 0.0
    Vouchers: Optional[list] = []
    PaymentTotals: Optional[list] = []

File: Vegtablity/Vegtablity/Models/Shift.vb
Namespace Models
    Public Class Shift
        Public Property ShiftID As Integer
        Public Property UserID As Integer
        Public Property UserName As String
        Public Property StartTime As DateTime
        Public Property EndTime As DateTime?
        Public Property StartingCash As Decimal
        Public Property EndingCash As Decimal?
        Public Property Status As String

        ' Summary properties from sp_Shift_GetSummary
        Public Property TotalSales As Decimal
        Public Property TotalPurchases As Decimal
        Public Property SalesCount As Integer
        Public Property PurchasesCount As Integer
        Public Property TotalPaidSales As Decimal
        Public Property TotalRemainder As Decimal
        Public Property TotalPaidPurchases As Decimal
        Public Property TotalPurchasesRemainder As Decimal
        Public Property TotalReceiptVouchers As Decimal
        Public Property TotalPaymentVouchers As Decimal
    End Class
End Namespace
